config/operator/rbac/auth_proxy_client_clusterrole.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
What is the `auth_porxy` and how relevant to our use case? 
Could it be something that kubebuilder creates for other use cases? For example for an Authorization hook or something else?

I think we need to understand what these YAMLs are for and only generate/include the ones needed by us.
IMHO, there are more objects than we really need.